app/services/proxy_test_service.rb

```diff
@@ -69,7 +69,7 @@ def initialize(proxy)
   end
 
   def disabled?
-    !proxy&.deployable?
+    proxy&.service_mesh_integration? || !proxy&.deployable?
```

Comment:
I think that the `nil` check is only needed once 😄, like: `!proxy || proxy.service_mesh_integration? || !proxy.deployable?`